lms/djangoapps/instructor/views/api.py

```diff
@@ -2833,52 +2826,62 @@ def update_forum_role_membership(request, course_id):
         which is limited to instructors.
     No one can revoke an instructors FORUM_ROLE_ADMINISTRATOR status.
 
-    Query parameters:
-    - `email` is the target users email
-    - `rolename` is one of [FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_GROUP_MODERATOR,
-        FORUM_ROLE_MODERATOR, FORUM_ROLE_COMMUNITY_TA]
-    - `action` is one of ['allow', 'revoke']
     """
-    course_id = CourseKey.from_string(course_id)
-    course = get_course_by_id(course_id)
-    has_instructor_access = has_access(request.user, 'instructor', course)
-    has_forum_admin = has_forum_access(
-        request.user, course_id, FORUM_ROLE_ADMINISTRATOR
-    )
+    permission_classes = (IsAuthenticated, permissions.InstructorPermission)
+    permission_name = permissions.EDIT_FORUM_ROLES
+    serializer_class = UpdateForumRoleMembershipSerializer
 
-    unique_student_identifier = request.POST.get('unique_student_identifier')
-    rolename = request.POST.get('rolename')
-    action = request.POST.get('action')
+    @method_decorator(ensure_csrf_cookie)
+    def post(self, request, course_id):
+        """
+        Handles role modification requests for a forum user.
 
-    # default roles require either (staff & forum admin) or (instructor)
-    if not (has_forum_admin or has_instructor_access):
-        return HttpResponseBadRequest(
-            "Operation requires staff & forum admin or instructor access"
+        Query parameters:
+        - `email` is the target users email
+        - `rolename` is one of [FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_GROUP_MODERATOR,
+            FORUM_ROLE_MODERATOR, FORUM_ROLE_COMMUNITY_TA]
+        - `action` is one of ['allow', 'revoke']
+        """
+        course_id = CourseKey.from_string(course_id)
+        course = get_course_by_id(course_id)
+        has_instructor_access = has_access(request.user, 'instructor', course)
+        has_forum_admin = has_forum_access(
+            request.user, course_id, FORUM_ROLE_ADMINISTRATOR
         )
```

Comment:
Is this a shortcoming of the permissions.InstructorPermission class? that we can't check these two and the above `EDIT_FORUM_ROLES` permissions at the same time for a user?  Could we adapt the permission class to check for multiple roles or use a different one so that all the permissions checks happen as DRF permission checks?

Reply:
@feanil in separate API which is related with forums I have re-write the functionality. 
Related [PR]
(https://github.com/openedx/edx-platform/pull/35366). I will merge `35366` first and then I will rebase this later.

`has_instructor_access` means user must be instructor.
`has_forum_admin` means user be added as `Administrator` in `Django_Comment_Common > Roles`.